src/Octoshift/VersionChecker.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OctoshiftCLI.Extensions;
+
+namespace OctoshiftCLI
+{
+    public class VersionChecker : IVersionProvider
+    {
+        private string _latestVersion;
+
+        public async Task<bool> IsLatest()
+        {
+            var curVersion = GetCurrentVersion();
+            var latestVersion = await GetLatestVersion();
+
+            curVersion = curVersion[..latestVersion.Length];
+
+            return curVersion == latestVersion;
+        }
+
+        public string GetCurrentVersion()
+        {
+            return Assembly.GetExecutingAssembly().GetName().Version.ToString();
+        }
+
+        public async Task<string> GetLatestVersion()
+        {
+            if (_latestVersion.IsNullOrWhiteSpace())
+            {
+                using var client = new HttpClient();
+                client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+                client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("OctoshiftCLI", GetCurrentVersion()));
```

Comment:
I was avoiding putting this in GithubApi because I think it will make things messy because the target GH for this call is always github.com, even if the source/target github instances are something different, the version check will always go to github.com, and it shouldn't have any PAT token set on this api call.

I'll make the change to inject HttpClient though, that makes sense.

Reply:
done